### DISCUSSION TITLE
Welcome to e-invoice-eu Discussions!

### DISCUSSION THREAD

**Discussion by gflohr** [gflohr/e-invoice-eu#29] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by JH-Dorn** [comment]
Hello, 
Im new to coding but im trying to understand it as best as I can. But if i would want to implement this as an app what would I need to do, like copy everything into an app folder and access that on the website or only the server files.
Ty in advance

**Comment by gflohr** [comment]
Sorry, for the late reply, long weekend ...

What app do you have in mind exactly? A mobile app?

In any case, you can have a look at the command-line tool. It gives you a pretty good idea on how to use the library, for example here: https://github.com/gflohr/e-invoice-eu/blob/main/apps/cli/src/commands/invoice.ts

Search for `doRun()`. It is the entry point into the tool, when you have given the command `e-invoice-eu cli`.

**Reply by JH-Dorn** [reply]
All good,

I'm trying to create a web-application like your example website, the difference would be that someone would be able to put all the data into the website and then it converts it to jsn and to an actual invoice without the need of another spreadsheet and mapping file. Is smth like that feasable from your standpoint out.

Ty for an answer I'm gonna look into it.

A little Visualisation of what the goal would look like:

![Screenshot_E-Invoice_Webexample](https://github.com/user-attachments/assets/ca6b3bd9-3020-4ff6-8656-495f639b14d5)

**Reply by gflohr** [reply]
Sure, that would be possible.

Some remarks:

There is no selection for the e-invoice format (ZUGFeRD, XRechnung, ...). But it has an impact on the UI. If the format is a hybrid format (ZUGFeRD/Factur-X) you always need a PDF. That will pose some specific problems. If the application is running the browser, then the PDF has to be uploaded and that means that your users will have to create it beforehand. That will make your application less useful. If you also have a backend, it would be easier because you can invoke other programs.

What you can also do is to create the PDF on the fly from the data that is input. There are quite a few solutions for creating PDFs with JavaScript, but I don't know any that is really easy to use. You could create HTML and try to convert the HTML to a PDF. But you will have to tweak the design, especially, if the invoice does not fit on one single page.

The other possiblity is to not offer ZUGFeRD/Factur-X at all. Then you have no problem.

Also keep in mind that the ZUGFeRD/Factur-X generation has a regression. These documents currently do not validate. The fix is already in a branch (the one that adds support for credit notes) and I will merge it soon. Have a look into #137! It will also make clear that you need a selection for the document type (invoice, corrected invoice, credit note, ...).
